plugins/chat/assets/javascripts/discourse/components/mention-warnings.js

```diff
@@ -0,0 +1,142 @@
+import Component from "@glimmer/component";
+import I18n from "I18n";
+import { escapeExpression } from "discourse/lib/utilities";
+import { htmlSafe } from "@ember/template";
+import { inject as service } from "@ember/service";
+
+export default class MentionWarnings extends Component {
+  @service siteSettings;
+  @service currentUser;
+  tagName = "";
+
+  get show() {
+    return (
+      this.args.tooManyMentions ||
+      this.args.unreachableGroupMentions.length > 0 ||
+      this.args.overMembersLimitGroupMentions.length > 0
+    );
+  }
+
+  get listStyleClass() {
+    if (this.args.tooManyMentions) {
+      return "chat-mention-warnings-list-simple";
+    }
+
+    if (
+      this.args.unreachableGroupMentions.length +
```

Comment:
We might want to have various helpers like `hasUnreachableGroupMentions` to avoid all these `this.args.fooBar.length`.

Should also probably be `this.args.fooBar?length`, we have no guarantee to have this arg.